src/vs/workbench/contrib/terminal/browser/terminalProfileResolverService.ts

```diff
@@ -315,18 +344,18 @@ export abstract class BaseTerminalProfileResolverService implements ITerminalPro
 		}
 	}
 
-	private _guessProfileIcon(shell: string): string | undefined {
+	private _guessProfileIcon(shell: string): ThemeIcon | undefined {
 		const file = path.parse(shell).name;
 		switch (file) {
 			case 'bash':
-				return Codicon.terminalBash.id;
+				return { id: Codicon.terminalBash.id };
 			case 'pwsh':
 			case 'powershell':
-				return Codicon.terminalPowershell.id;
+				return { id: Codicon.terminalPowershell.id };
 			case 'tmux':
-				return Codicon.terminalTmux.id;
+				return { id: Codicon.terminalTmux.id };
 			case 'cmd':
-				return Codicon.terminalCmd.id;
+				return { id: Codicon.terminalCmd.id };
```

Comment:
For these I think we can just refer to the Codicon directly:

```
return Codicon.terminalBash;
```

Reply:
We can't bc Codicon is no longer a valid return type